### PR TITLE
Error Messages for Incorrect Login Attempts #38

### PR DESCRIPTION
**Pull request by KartikeySharma**

### CODE REVIEW

File: src/accounts/backends.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        # Use user_can_authenticate method from ModelBackend
+        if user.check_password(password) and self.user_can_authenticate(user):
+            return user
+        return None
+
+    def get_user(self, user_id):
+        UserModel = get_user_model()
+        try:
+            return UserModel.objects.get(pk=user_id)
+        except UserModel.DoesNotExist:
```
Comment (ahhcash) [inline src/accounts/backends.py]: this looks good but I think we can just have username based auth for now. we can add a unique constraint on the email on the CustomUser model, and just fetch the username based on the email. i think that would be easier right?

File: src/accounts/forms.py [hunk 2]
```diff
@@ -13,37 +14,77 @@ class Meta:
...
-                raise forms.ValidationError("This page is for service providers only.")
-        except UserModel.DoesNotExist:
-            raise forms.ValidationError("Invalid email or password")
-        return cleaned_data
+        email = self.cleaned_data.get("email")
+        password = self.cleaned_data.get("password")
+
+        if email and password:
+            self.user_cache = authenticate(self.request, email=email, password=password)
+            if self.user_cache is None:
```
Comment (ahhcash) [inline src/accounts/forms.py:74]: we could just fetch the username based on the email and perform a normal username based auth here. that's what i did for #87. you can look at the changes there (to add a unique constraint for the email field)

**Review by ahhcash — CHANGES_REQUESTED** [review]
few comments. also please add some UTs to cover these changes 😉

**Comment by ahhcash** [comment]
also merge conflicts. I'll work on moving the current dev environemnt's sqlite3 to a managed solution for #61

**Review by omerfarukbasar — APPROVED** [review]
Looks good to me!

**Review by ahhcash — APPROVED** [review]
lgtm

**Pull request merged**